### ISSUE TITLE
Degraded performance due to excessive disk (.parcel-cache) and memory performance?

### ISSUE THREAD

**Issue by broofa** [parcel-bundler/parcel#6236]
# 🐛 bug report

[Note: Probably related to #5072 but opening as a new issue due to `.parcel-cache` being a possible factor]

I keep the parcel dev server (`parcel index.html`) running in the background.  After a day or so of active development, it eventually stops responding.   Ctrl-C'ing the process does not work or, if it does, it can take a *long* time to work. Sometimes a minute or more.   

Most times I'll just `kill -9` the process, after which I'll `rm -fr .parcel-cache` before restarting.

This is not a particularly large project.   ~2000 lines of TypeScript and  9-10 direct NPM dependencies (the only "big" ones being react, firebase, and bootstrap). The repository is at https://github.com/broofa/flightcard if that's helpful.

The last time this occurred .parcel-cache had ~5000 files in it, weighting in at 4.3GB:
```
$ find .parcel-cache/ | wc -l
    4861

$ du -sh .parcel-cache
4.3G	.parcel-cache
```

I'm not sure if that's expected or not, but it seems... excessive.  And, given that deleting the cache appears to "fix" the problem for a time, it would seem to suggest the cache is a contributing factor here.

## 🎛 Configuration (.babelrc, package.json, cli command)

I don't have much in the way of configuration files.  package.json can be found [here](https://github.com/broofa/flightcard/blob/main/package.json).

## 💁 Possible Solution

Delete .parcel-cache.

## 🔦 Context

This is occuring on my laptop, which is a fully-loaded 2020 13" Macbook Air M1 (see specs below).  I.e. There should be plenty of compute horsepower available.

While the problem is obviously not a showstopper, it is disruptive to development flow.  Performance will gradually deteriorate, causing hot-module reloads to take progressively longer.   Eventually it gets annoying enough I'm compelled to restart the process.

## 💻 Code Sample

See repository link above.  Sorry, I know that's probably not enough to reproduce the problem, but this is one of those bugs that's hard to articulate a precise recipe for.

## 🌍 Your Environment

```
$ npx envinfo --system --browsers --npmPackages --binaries
npx: installed 1 in 0.921s

  System:
    OS: macOS 11.1
    CPU: (8) x64 Apple M1
    Memory: 50.92 MB / 16.00 GB
    Shell: 5.1.0 - /usr/local/bin/bash
  Binaries:
    Node: 14.16.1 - ~/.nvm/versions/node/v14.16.1/bin/node
    npm: 6.14.12 - ~/.nvm/versions/node/v14.16.1/bin/npm
  Browsers:
    Chrome: 90.0.4430.93
    Firefox: 88.0
    Safari: 14.0.2
  npmPackages:
    @parcel/babel-plugin-transform-runtime: 2.0.0-nightly.1823 => 2.0.0-nightly.1823
    @types/react-dom: 17.0.3 => 17.0.3
    @types/react-router-dom: 5.1.7 => 5.1.7
    @typescript-eslint/eslint-plugin: 4.20.0 => 4.20.0
    @typescript-eslint/parser: 4.20.0 => 4.20.0
    bootstrap: 4.6.0 => 4.6.0
    eslint: 7.21.0 => 7.21.0
    eslint-config-standard: 16.0.2 => 16.0.2
    eslint-plugin-import: 2.22.1 => 2.22.1
    eslint-plugin-node: 11.1.0 => 11.1.0
    eslint-plugin-promise: 4.3.1 => 4.3.1
    eslint-plugin-react: 7.22.0 => 7.22.0
    firebase: 8.4.2 => 8.4.2
    firebaseui: 4.8.0 => 4.8.0
    nanoid: 3.1.22 => 3.1.22
    parcel: 2.0.0-beta.1 => 2.0.0-beta.1
    react: ^16.13.1 => 16.14.0
    react-bootstrap: 1.5.2 => 1.5.2
    react-dom: ^16.13.1 => 16.14.0
    react-router-dom: 5.2.0 => 5.2.0
    sass: 1.32.12 => 1.32.12
    typescript: 4.2.3 => 4.2.3
```

**Comment by mischnic** [comment]
> .parcel-cache had ~5000 files in it, weighting in at 4.3GB:

We are aware of that and cache garbage collection is planned (but cache correctness/invalidation is the priority right now).

I have never noticed that this actually causes slower builds over time myself though

**Comment by mischnic** [comment]
> This is not a particularly large project. ~2000 lines of TypeScript and 9-10 direct NPM dependencies (the only "big" ones being react, firebase, and bootstrap). The repository is at broofa/flightcard if that's helpful.

Looks like that repo is private or has a different name? (But the problem is probably not specific to your project)

I'll try to create a script to simulate many edits in serve mode without restarting in a different project.

**Comment by aminya** [comment]
I am also having an issue on Windows that the build never finishes. Reproducible here:
https://github.com/atom-community/terminal

**Comment by mischnic** [comment]
@aminya Your problem might be https://github.com/parcel-bundler/parcel/issues/6138 instead

**Comment by broofa** [comment]
> Looks like that repo is private

@mischnic Fixed.  ("Dooh!")

**Comment by andys8** [comment]
Can report a similar issue: `data.mdb` is 5GB in size.

```
  File: data.mdb
  Size: 5598126080	Blocks: 10933856   IO Block: 4096   regular file
Device: 254,0	Inode: 15366414    Links: 1
Access: (0644/-rw-r--r--)  Uid: ( 1000/ andreas)   Gid: ( 1000/ andreas)
Access: 2022-07-25 23:13:15.147203863 +0200
Modify: 2022-08-09 13:32:02.780086584 +0200
Change: 2022-08-09 13:32:02.780086584 +0200
 Birth: 2022-07-25 23:13:15.147203863 +0200
```

**Comment by broofa** [comment]
Closing as I haven't seen this issue since upgrading to `parcel@2.6.2`.  I've also upgraded laptops and have been actively developing the flightcard repo I reference in the original issue, so my environment has changed enough I wouldn't be able to effectively reproduce / test any fixes that might be made here.

@andys8 'Might make sense for you to file a new issue with more details and, most especially, a detailed recipe for reproducing your issue.
